KaitaiStream.cs

```diff
@@ -7,67 +7,89 @@
 namespace Kaitai
 {
     /// <summary>
-    /// The base Kaitai steam which exposes an API for the Kaitai Struct framework.
+    /// The base Kaitai stream which exposes an API for the Kaitai Struct framework.
     /// It's based off a <code>BinaryReader</code>, which is a little-endian reader.
     /// </summary>
     public partial class KaitaiStream : BinaryReader
     {
         #region Constructors
 
+        /// <summary>
+        /// Create a KaitaiStream backed by abstract stream. It could be in-memory buffer or open file.
+        /// </summary>
         public KaitaiStream(Stream stream) : base(stream)
         {
         }
 
-        ///<summary>
-        /// Creates a KaitaiStream backed by a file (RO)
-        ///</summary>
-        public KaitaiStream(string file) : base(File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read))
+        /// <summary>
+        /// Create a KaitaiStream backed by a closed file (in read-only binary-mode).
+        /// </summary>
+        public KaitaiStream(string filename) : base(File.Open(filename, FileMode.Open, FileAccess.Read, FileShare.Read))
         {
         }
 
-        ///<summary>
-        ///Creates a KaitaiStream backed by a byte buffer
-        ///</summary>
-        public KaitaiStream(byte[] bytes) : base(new MemoryStream(bytes))
+        /// <summary>
+        /// Create a KaitaiStream backed by in-memory buffer.
+        /// </summary>
+        public KaitaiStream(byte[] data) : base(new MemoryStream(data))
         {
         }
 
+        /// <summary>
+        /// Used internally.
+        /// </summary>
         private ulong Bits = 0;
+        /// <summary>
+        /// Used internally.
+        /// </summary>
         private int BitsLeft = 0;
 
+        /// <summary>
+        /// Used internally.
+        /// Caches the current plaftorm endianness and allows emited bytecode to be optimised. Thanks to @Arlorean.
+        /// </summary>
         static readonly bool IsLittleEndian = BitConverter.IsLittleEndian;
 
+        static KaitaiStream()
+        {
+            compute_single_rotations();
+        }
+
         #endregion
 
         #region Stream positioning
 
         /// <summary>
-        /// Check if the stream position is at the end of the stream
+        /// Check if the stream position is at the end of the stream (at EOF).
+        /// WARNING: This requires a seekable and tellable stream.
```

Comment:
.NET does not use "tell" and thus you can expect that nobody would knows what "tellable" is. Microsoft actually calls both "seekable" and "tellable" => ["supports seeking"](https://msdn.microsoft.com/en-us/library/system.io.stream.seek(v=vs.110).aspx), and that actually makes sense to use that wording at least here.

Reply:
In our own documentation (KS docs PR regarding seeking) we do use these terms, seekable and tellable, so there is no reason why we cant make it consistent with our own terminology, instead of Microsoft's. And I do keep Microsoft .NET in high regard. Its also consistent with docstrings in Python and in future, other runtimes.

Reply:
Hey, dont yell at me. If you are adamant that it should be changed, then be explicit. You only old me your opinion on this, and I told you mine. I will fix it.

Reply:
Fixed.

Reply:
I certainly don't want to yell at anyone or anything like that. It just looked like we had no discussion there and you just kept pushing that C/Python jargonism where it does not belong (and that would have raised lots of questions in future). I've stated my position in https://github.com/kaitai-io/kaitai_struct_doc/pull/14, and you've just ignored my comment, continuing to use it here and there.

Reply:
Sorry, I didnt make the connection. That and just got overall sloppy... :sleepy: